battle-engine.js

```diff
@@ -2368,6 +2369,11 @@ Battle = (function () {
 				statuses.push({status: status, callback: status[callbackType], statusData: this.formatData, end: function () {}, thing: thing, priority: status[callbackType + 'Priority'] || 0});
 				this.resolveLastPriority(statuses, callbackType);
 			}
+			if (this.events && this.events[callbackType] !== undefined) {
+				for (var i = 0; i < this.events[callbackType].length; i++) {
+					statuses.push({status: status, callback: this.events[callbackType][i].callback, statusData: this.formatData, end: function () {}, thing: thing, priority: this.events[callbackType][i].priority});
+				}
```

Comment:
`resolveLastPriority` is missing here.

I guess adding it will require adding strongly negative priorities to test event handlers?